google/cloud/forseti/services/inventory/base/resources.py

```diff
@@ -849,6 +850,40 @@ class AppEngineInstance(resource_class_factory('appengine_instance', 'name',
 class BigqueryDataSet(resource_class_factory('dataset', 'id')):
     """The Resource implementation for Bigquery DataSet."""
 
+    def _set_cache(self, field_name, value):
+        """Manually set a cache value if it isn't already set.
+
+        Args:
+            field_name (str): The name of the attribute to cache.
+            value (str): The value to cache.
+        """
+        field_name = '__cached_{}'.format(field_name)
+        if not hasattr(self, field_name) or getattr(self, field_name) is None:
+            setattr(self, field_name, value)
+
+    @cached('iam_policy')
+    def get_iam_policy(self, client=None):
+        """IAM policy for this Dataset.
+
+        Args:
+            client (object): GCP API client.
+
+        Returns:
+            dict: Dataset Policy.
+        """
+        try:
+            iam_policy = client.fetch_bigquery_iam_policy(
+                self.parent()['projectNumber'],
+                self['datasetReference']['datasetId'])
+            dataset_policy = iam_helpers.convert_iam_to_bigquery_policy(
+                iam_policy)
+            self._set_cache('dataset_policy', dataset_policy)
```

Comment:
qq: Can we assume that only one of IAM or BigQuery Access Policy is available at at time?

Reply:
Yes, the API only returns an access policy, and CAI only returns an IAM policy. It's a weird edge case where the API isn't exposing the IAM policy at this time, although the backend is converting the access policy to an IAM policy automatically.